src/Features/Core/Portable/Completion/CompletionOptions.cs

```diff
@@ -199,4 +199,17 @@ public Metadata()
                     storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.ProvideDateAndTimeCompletions"));
         }
     }
+
+    internal static class CompletionOptionsExtensions
+    {
+        public static bool ShouldShowItemsFromUnimportNamspaces(this CompletionOptions options)
+        {
+            // Don't trigger import completion if the option value is "default" and the experiment is disabled for the user. 
+            var importCompletionOptionValue = options.ShowItemsFromUnimportedNamespaces;
+            if (importCompletionOptionValue == false || (importCompletionOptionValue == null && !options.TypeImportCompletion))
+                return false;
```

Comment:
this is just confusing :)  please break into two statements.  haveing ||'s and &&'s and !'s breaks my brain.

Reply:
Isn't this equivalent to

`return options.ShowItemsFromUnimportedNamespaces ?? options.TypeImportCompletion`